File: js/src/gilded_rose.js
class Shop {
  constructor(items = []) {
    this.items = items;
  }
  updateQuality() {

    this.items.forEach(function (item) {
      if (item.name != 'Sulfuras, Hand of Ragnaros') {
        item.sellIn = item.sellIn - 1;
      }

      item.updateQuality();
    })

    return this.items;
  }
}

File: Kotlin/src/main/kotlin/com/gildedrose/GildedRose.kt
package com.gildedrose

import com.gildedrose.processor.ItemProcess


class GildedRose(var items: Array<Item>) {

    fun updateQuality() {
        items = items.map { item ->
            ItemProcess(name = item.name, quality = item.quality, sellIn = item.sellIn).let { itemData ->
                itemData.process().toItem()
            }
        }.toTypedArray()
    }

}


